basket/main.py

```diff
@@ -46,12 +71,26 @@ class Basket(object):
     # gets out of control.
     os = os
     tarfile = tarfile
-    urllib = urllib
+    urllib = urllib2
     zipfile = zipfile
     err = sys.stderr
     out = sys.stdout
     root = os.environ.get('BASKET_ROOT') or os.path.expanduser('~/.basket')
 
+    # Support http_proxy environment
+    # because of usage xmlrpclib and urllib we need to support both mechanisms
+    if os.environ.get('http_proxy'):
+        # Extract http_proxy from environment
+        proxy = ''.join(re.findall(r'[\d\.]+:?', os.environ.get('http_proxy')))
```

Comment:
Oh, I think I got it. Are you in fact looking for an IP address and trying to remove the protocol from the `http_proxy`  environment variable? That works fine if the environment variable looks like `http://10.1.2.3:1234`. However, I suppose that the environment variable may look like `http://proxy:1234`, too. If so, your regexp fails. Perhaps you should rather try to remove the protocol [1]? You can use `urlparse.urlparse()` (moved to `urlib.urlparse()` in Python 3.

[1] But do you really need to remove the protocol? I have never had to setup a proxy transport like this, I have no idea. :)